taskwiki/vwtask.py

```diff
@@ -360,27 +360,26 @@ def find_parent_task(self):
     def apply_defaults(self):
         from taskwiki import viewport, preset
 
+        # Find first parent header or viewport
+        header, port = None, None
         for i in reversed(range(0, self['line_number'])):
             header = preset.PresetHeader.from_line(i, self.cache)
+            port = viewport.ViewPort.from_line(i, self.cache)
```

Comment:
Premature optimization is root of all evil, but faint memories of trying to optimize the number of times we are trying to initialize a ViewPort, because of tangible performance effect.

That said, it should live in the cache, which should ensure it is evaluated only once per buffer. So I guess this should be fine :thinking:

Reply:
If I could use `:=` I probably would but that's a recent addition to Python so I can't. :-)

Reply:
I actually had the same thought (currently running F32/Python 3.8.3), but it might be too early for the more conservative part of the Linux universe :)